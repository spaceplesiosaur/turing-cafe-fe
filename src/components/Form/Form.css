.form {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.formInput {
  width: 15%;
  padding: 1em;
  border: 1px solid black
}

.formButton {
  width: 15%;
  padding: 1em;
  border: 1px solid black;
  background-color: teal
}
