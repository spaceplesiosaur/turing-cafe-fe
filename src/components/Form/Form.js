import React, { Component } from 'react'
import './Form.css'

export default class Form extends Component {
  constructor() {
    super()
    this.state = {
      name: "Name",
      date: "Date (mm/dd)",
      time: "Time",
      guests: "Number of Guests"
    }
  }

  handleChange = (event) => {
    this.setState({[event.target.name]: event.target.value})
  }

  setUser = () => {
    this.props.addTable({name: this.state.name, date: this.state.date, time: this.state.time, number: this.state.guests})
  }

  render() {
    return (
      <section className="form">
        <input className="formInput" id="nameInput" name="name" value={this.state.name} onChange={this.handleChange}></input>
        <input className="formInput" id="dateInput" name="date" value={this.state.date} onChange={this.handleChange}></input>
        <input className="formInput" id="timeInput" name="time" value={this.state.time} onChange={this.handleChange}></input>
        <input className="formInput" id="guestsInput" name="guests" value={this.state.guests} onChange={this.handleChange}></input>
        <button className="formButton" onClick={this.setUser}>Make Reservation</button>
      </section>
    )
  }
}
