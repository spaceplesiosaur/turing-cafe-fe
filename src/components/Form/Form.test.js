import React from 'react';
import { shallow } from 'enzyme'
import Form from './Form';

describe('Form', () => {

  it('should match the snapshot with correct data being pulled in', () => {
    const wrapper = shallow(
      <Form
      addTable={jest.fn()}
      />)

    expect(wrapper).toMatchSnapshot();
  })

  it('should call setUser with the correct argument', () => {
    const mockTable = {
      name: "Taylor Swift",
      date: "08/22",
      time: "5:00",
      number: "13"
    }

    const mocksetUser = jest.fn()

    const wrapper = shallow(
      <Form
      addTable={mocksetUser}
      />)

    wrapper.find(".formButton").simulate('click')
    expect(mocksetUser).toHaveBeenCalled()

  })

  it('should call handleChange with correct argument when imput is changed', () => {
    const mockName = "The Rock"
    const mockEvent = {event: {target: {name: "name", value: "The Rock"}}}
    const mockHandleChange = jest.fn()


    const wrapper = shallow(
      <Form
      addTable={jest.fn()}
      />)

    wrapper.find("#nameInput").simulate('change', mockEvent)

    expect(mockHandleChange).toHaveBeenCalledWith(mockEvent)

  })
})
