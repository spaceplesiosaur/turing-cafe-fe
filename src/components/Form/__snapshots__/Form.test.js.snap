// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with correct data being pulled in 1`] = `
<section
  className="form"
>
  <input
    className="formInput"
    id="nameInput"
    name="name"
    onChange={[Function]}
    value="Name"
  />
  <input
    className="formInput"
    id="dateInput"
    name="date"
    onChange={[Function]}
    value="Date (mm/dd)"
  />
  <input
    className="formInput"
    id="timeInput"
    name="time"
    onChange={[Function]}
    value="Time"
  />
  <input
    className="formInput"
    id="guestsInput"
    name="guests"
    onChange={[Function]}
    value="Number of Guests"
  />
  <button
    className="formButton"
    onClick={[Function]}
  >
    Make Reservation
  </button>
</section>
`;
